### PR TITLE
Allow event triggers to be removed with the callback stored.

### PR DESCRIPTION
**Pull request by aaronpcz**
Test implemented demonstrating the bug.
.on('evt', fnt) and if that fnt implements off('evt') it would not work.  This has been fixed.

For most use cases using once is enough, but this does not cover every possible way to use on and off for event subscriptions.

A previous pull request did not modify the source and only the build output.
737 /src/audio5.js
957 /src/audio5.js

Without the source being updated, when the grunt task is ran, the changes are reverted (this has been fixed as part of this commit)

### CODE REVIEW

File: src/audio5.js [hunk 1]
```diff
@@ -723,8 +735,9 @@
      */
     play: function () {
       if(this.audio) {
-        this.audio.play();
+        var playPromise = this.audio.play();
```
Comment (aaronpcz) [inline src/audio5.js]: Only changed in /src/audio5 as it looks like previous change was made to build output rather than src

File: src/audio5.js [hunk 2]
```diff
@@ -941,7 +954,7 @@
      */
     play: function () {
       if(!this.playing){
-        this.audio.play();
+        return this.audio.play();
```
Comment (aaronpcz) [inline src/audio5.js]: Only changed in /src/audio5 as it looks like previous change was made to build output rather than src

File: audio5.js [hunk 3]
```diff
@@ -134,17 +134,29 @@
...
         while(this.channels[evt].length > 0) {
           var sub = this.channels[evt].shift();
-          if (typeof (sub.fn) === 'function') {
-            sub.fn.apply(sub.ctx, args);
-          }
+
           if ( !sub.once ){
-            a.push(sub);
+            // for any channel event triggers that are not marked as once only, we need to keep.
+            this.channels[evt].push(sub);
```
Comment (zohararad) [inline audio5.js]: Won't this line cause an infinite loop? If you shift the subscribe from the channel on line 142 and then push it again on line 146, the while loop in line 141 might never exit

Comment (aaronpcz) [reply audio5.js]: Yep - this was so shoddy refactoring on my behalf, I've opened a much more respectable pull request.  Apologies for your time used looking at this.

Comment (zohararad) [reply audio5.js]: No worries.. Let me know when I can review the code again

Comment (aaronpcz) [reply audio5.js]: https://github.com/zohararad/audio5js/pull/96

Here we go

**Pull request closed without merging**